File: 2017/02/process.lua
local input = io.read("*all"):gsub("^%s*(.-)%s*$", "%1")

local sum_step1=0
local sum_step2=0

for line in input:gmatch("[^\n]+") do
  local n = {}
  for num in line:gmatch("%S+") do
      table.insert(n, tonumber(num))
  end  

  local max,min = tonumber(n[1]), n[1]

  for _,v in ipairs(n) do
    if v>max then max=v end
    if v<min then min=v end
  end

  local diff = max-min
  sum_step1 = sum_step1 + diff

  for _,n1 in ipairs(n) do
    for _,n2 in ipairs(n) do
      if (n1~=n2) and (n1%n2==0) then
        sum_step2 = sum_step2+n1/n2
      end
    end
  end
end


print("Sum step1:", sum_step1)
print("Sum step2:", sum_step2)

File: 2017/02/process.js
const input = require('fs').readFileSync(0).toString().trim(); 
let sum_step1=0;
let sum_step2=0;
for (const line of input.split("\n")) {
  const n=line.split(/\s+/).map(Number);
  const max=Math.max(...n)
  const min=Math.min(...n)
  const diff=max-min;
  sum_step1+=diff;

  console.log(n);
  for (const n1 of n)
    for (const n2 of n)
      if ((n1!==n2) && (n1%n2==0))
        sum_step2+=n1/n2;

};
console.log(sum_step1, sum_step2);
